src/loaders/secretsManager.ts

```diff
@@ -37,7 +41,10 @@ function getEnvVarValueOrFail(metadata: Metadata, envVarName: string) {
   return value;
 }
 
-export default async function loadFromSecretsManager(metadata: Metadata) {
+export default async function loadFromSecretsManager(
+  metadata: Metadata,
+  clientOverride?: SecretsManagerClient
```

Comment:
just a means to allow the testing to inject a client

Reply:
hmm just looked again and we need to make sure the mocked client is the exact one it uses, not sure we can do that without passing it in directly.

and this lib isn't changed that much so we might be alright